### ISSUE TITLE
foreign key dropdown doesn't handle citext on postgres

### ISSUE THREAD

**Issue by MolallaComm** [HeidiSQL/HeidiSQL#1931]
When editing int identity columns, in addition to the identity key, Heidi also usually displays the first varchar or text column in hopes of making things more human readable.  I noticed that if the FK table is using "citext" instead of "text" columns it doesn't work (the identity numbers still show up in the dropdown but not the human readable part).  If I change the column from "citext" to "text" and refresh, it works as expected.

Unfortunately, using case insensitive coalations even in this day and age with postgres is often problematic for many reasons so we and many others use citext to work around these limitations.  Would be nice if Heidi worked for citext as well as varchar and text on postgres.

**Comment by ansgarbecker** [comment]
I was just trying to create a test table with a CITEXT column, but to my surprise, on Postgesql 15.1, I get an SQL error saying the type "CITEXT" does not exist:

![grafik](https://github.com/HeidiSQL/HeidiSQL/assets/7986591/8f01dcd3-a91b-4c2c-9751-a1f5d666f8a6)

[CITEXT Documentation](https://www.postgresql.org/docs/current/citext.html) says it should work from PostgreSQL 12.0+ so it should work on my local server.

But the documentation also says something about a module installation, and I did not install any module. I have no clue what that means. Can you shed some light on that?

**Comment by ansgarbecker** [comment]
Thanks for the tip. That did it.

`CITEXT` is now supported in the next builds, and categorized as text type, so it shows up in the foreign value pulldown you mentioned.

![grafik](https://github.com/HeidiSQL/HeidiSQL/assets/7986591/a16b696b-9918-4a47-990c-b6776980908e)

Note the value in `IS.columns.data_type` is "USER-DEFINED" for such extension based types. I worked around that by using the value in `udt_name`, which was the correct "citext" in my test case.

I'm crossing fingers the hardcoded oid `24651` is unique everywhere, and not only on my local server.

**Comment by MolallaComm** [comment]
Yes - unfortunately - I think the oid changes from server to server/database to database.  I was recently doing some similar work and noticed on my machine that citext was 2606673 but on the production server it was 59586 - i think the built in types are always the same, but ones from extensions like citext will vary.  Hopefully that won't throw a wrench in things - if you want, i can download tomorrows build and verify for sure that it doesn't work, but I'm reasonably confident it won't if it is counting on 24651 being associated with citext.

**Comment by ansgarbecker** [comment]
Yes, that would be helpful if you give it a test and give some feedback. The new build should be available in roughly half an hour.

**Comment by ansgarbecker** [comment]
Similar issue in a different project: https://github.com/launchbadge/sqlx/issues/295

**Comment by MolallaComm** [comment]
Yep - not sure if it helps, but the SQL to get it seems to be:

SELECT 'citext'::regtype::oid;

and I did confirm it is different on every server I tried.  Also, this article discusses getting it via the C api:

https://stackoverflow.com/questions/45887484/get-user-defined-type-oid-in-postgresql-extension-c
